File: OOP/tic-tac-toe-board-game-app/src/com/techlab/board/Cell.java
package com.techlab.board;

import com.techlab.Exception.AllReadyMarkCellException;

public class Cell {

	private Mark cell;

	public Cell() {
		cell = Mark.EMPTY;
	}

	public Mark getCellState() {
		return cell;
	}

	public void setCellState(Mark cell) throws AllReadyMarkCellException {
		
			if (!checkEmptyCell(cell))
				throw new AllReadyMarkCellException("cell is allready fill");
		
		this.cell = cell;
	}

	private boolean checkEmptyCell(Mark cell) {
		if (this.cell.equals(Mark.EMPTY))
			return true;
		return false;
	}
}

File: nodeJs/tic-tac-toe-board-game/cell.js
let mark = require('./mark');
let cellAllReadyMarkException = require('./cellAllReadyMarkException');

"use strict";
class Cell {
    constructor() {
        this._cell = mark.EMPTY;
    }
    set cellState(mark) {
        if (!this.checkEmptyCell(this._cell))
            throw new cellAllReadyMarkException("cell is allready fill");
        this._cell = mark;
    }
    get cellState() {
        return this._cell;
    }
    checkEmptyCell(cell) {
        if (cell == mark.EMPTY)
            return true;
        return false;
    }
}
module.exports = Cell;